src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/mapper/EnvelopeMapper.java

```diff
@@ -0,0 +1,107 @@
+
+public class EnvelopeMapper {
+
+    private EnvelopeMapper() {
+        // utility class
+    }
+
+    public static Envelope toDbEnvelope(InputEnvelope envelope, String containerName) {
+        return new Envelope(
+            envelope.poBox,
+            envelope.jurisdiction,
+            envelope.deliveryDate,
+            envelope.openingDate,
+            envelope.zipFileCreateddate,
+            envelope.zipFileName,
+            envelope.caseNumber,
+            envelope.classification,
+            toDbScannableItems(envelope.scannableItems),
+            toDbPayments(envelope.payments),
+            toDbNonScannableItems(envelope.nonScannableItems),
+            containerName
+        );
+    }
+
+    private static List<ScannableItem> toDbScannableItems(List<InputScannableItem> scannableItems) {
+        if (scannableItems != null) {
+            return scannableItems
+                .stream()
+                .map(EnvelopeMapper::toDbScannableItem)
+                .collect(toList());
+        } else {
+            return emptyList();
+        }
+    }
+
+    private static ScannableItem toDbScannableItem(InputScannableItem scannableItem) {
+        return new ScannableItem(
+            scannableItem.documentControlNumber,
+            scannableItem.scanningDate,
+            scannableItem.ocrAccuracy,
+            scannableItem.manualIntervention,
+            scannableItem.nextAction,
+            scannableItem.nextActionDate,
+            scannableItem.ocrData,
+            scannableItem.fileName,
+            scannableItem.notes,
+            scannableItem.documentType
+        );
+    }
+
+    private static List<Payment> toDbPayments(List<InputPayment> payments) {
+        if (payments != null) {
+            return payments
+                .stream()
+                .map(EnvelopeMapper::toDbPayment)
+                .collect(toList());
+        } else {
+            return emptyList();
+        }
+    }
+
+    private static Payment toDbPayment(InputPayment payment) {
+        return new Payment(
+            payment.documentControlNumber,
+            payment.method,
+            payment.amount,
+            payment.currency,
+            payment.paymentInstrumentNumber,
+            payment.sortCode,
+            payment.accountNumber
+        );
+    }
+
+    private static List<NonScannableItem> toDbNonScannableItems(List<InputNonScannableItem> nonScannableItems) {
+        if (nonScannableItems != null) {
+            return nonScannableItems
+                .stream()
+                .map(EnvelopeMapper::toDbNonScannableItem)
+                .collect(toList());
+        } else {
+            return emptyList();
+        }
+    }
+
+    private static NonScannableItem toDbNonScannableItem(InputNonScannableItem nonScannableItem) {
+        return new NonScannableItem(
+            nonScannableItem.documentControlNumber,
+            nonScannableItem.itemType,
+            nonScannableItem.notes
+        );
+    }
```

Comment:
What kind of simplification do you have in mind?

Reply:
each case is a conversion from one class to another. so why not
```java
NonScannableItem(InputNonScannableItem input) {
    // set all final fields in this constructor
}
```

Reply:
imho those 2 models should be independent.

Reply:
I see the benefit of having one argument instead of many, but that would happen at the cost of making one set of models aware of the other. Like Rafał, I think it's best to avoid it.